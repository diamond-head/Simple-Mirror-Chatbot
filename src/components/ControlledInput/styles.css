.controlled-input__container {
  width: 100%;
}

.controlled-input__textarea {
  resize: none;
  outline: none;
  padding: 16px;
  margin: 0;
  border: none;
  min-width: 100%;
  min-height: 50px;
  font-size: 14px;
  font-style: normal;
  font-weight: normal;
  box-shadow: none;
  letter-spacing: 0;
  color: var(--text-gray);
  line-height: 20px;
  width: auto;
  background: transparent;
  font-family: Georgia, 'Times New Roman', Times, serif;
}