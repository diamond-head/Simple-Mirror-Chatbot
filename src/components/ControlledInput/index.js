import React from 'react';

import "./styles.css";

export default function ControlledInput({ 
  name,
  placeholder,
  defaultValue,
  onValueChange,
  ...props
}) {
  return (
    <div className="controlled-input__container">
      <textarea
        className={"controlled-input__textarea"}
        name={name}
        rows={1}
        value={defaultValue}
        placeholder={placeholder}
        onChange={({ target: { value } }) => {
          onValueChange(value);
        }}
      />
    </div>
  );
}