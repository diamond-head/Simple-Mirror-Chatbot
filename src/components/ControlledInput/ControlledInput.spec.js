import React from 'react';
import renderer from 'react-test-renderer';

import ControlledInput from "./index";

describe("<ControlledInput /> component", () => {
  it("ControlledInput snapshot renders", () => {
    const component = renderer.create(
      <ControlledInput name="test-input" placeholder="Please enter message..." defaultValue="" onValueChange={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});