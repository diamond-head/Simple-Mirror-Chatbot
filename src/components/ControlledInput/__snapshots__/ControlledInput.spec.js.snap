// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlledInput /> component ControlledInput snapshot renders 1`] = `
<div
  className="controlled-input__container"
>
  <textarea
    className="controlled-input__textarea"
    name="test-input"
    onChange={[Function]}
    placeholder="Please enter message..."
    rows={1}
    value=""
  />
</div>
`;
